plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -17,149 +18,55 @@
-    options: Optional[List[str]] = None
-    cast: str = "float"
-    log: bool = False
-
-    def __post_init__(self) -> None:
-        if not (self.bounds is None) ^ (self.options is None):
-            raise ValueError("Exactly one of bounds or options must be specified")
-        if self.bounds is not None:
-            if self.cast == "str":
-                raise ValueError(
-                    "Inconsistent specifications 'str' for bounded values."
-                )
-            if self.bounds[0] > self.bounds[1]:
-                raise ValueError(f"Bounds must be ordered, but got {self.bounds}")
-        if self.options is not None and self.log:
-            raise ValueError("Inconsistent 'log' specification for choice parameter")
-
-    @classmethod
-    def parse(cls, string: str) -> "CommandlineSpec":
-        """Parses a commandline argument string
-
-        Parameter
-        ---------
-        string: str
-            This can be:
-             - comma-separated values: for a choice parameter
-               Eg.: "a,b,c"
-             - colon-separated values for ranges of scalars.
-               Eg.: "0:10"
-            Colon-separeted can be appended to:
-             - cast to int/str/float (always defaults to float):
-               Eg: "float:0,4,10", "int:0:10"
-             - set log distribution for scalars
-               Eg: "int:log:4:1024"
-        """
-        available_modifiers = {"log", "float", "int", "str"}
-        colon_split = string.split(":")
-        modifiers = set(
-            itertools.takewhile(available_modifiers.__contains__, colon_split)
-        )
-        remain = colon_split[len(modifiers) :]
-        casts = list(modifiers - {"log"})
-        if len(remain) not in {1, 2}:
-            raise ValueError(
-                "Can't interpret non-speficiations: {}.\nthis needs to be "
-                "either colon or coma-separated values".format(":".join(remain))
-            )
-        if len(casts) > 1:
-            raise ValueError(f"Inconsistent specifications: {casts}")
-        if len(remain) == 1:  # choice argument
-            cast = casts[0] if casts else "str"
-            options = remain[0].split(",")
-            if len(options) < 2:
-                raise ValueError("At least 2 options are required")
-            if not casts:
-                try:  # default to float if possible and no spec provided
-                    _ = [float(x) for x in options]
-                    cast = "float"
-                except ValueError:
-                    pass
-            return cls(options=options, cast=cast)
-        # bounded argument
-        bounds: Tuple[float, float] = tuple(float(x) for x in remain)  # type: ignore
-        cast = casts[0] if casts else "float"
-        return cls(bounds=bounds, cast=cast, log="log" in modifiers)
-
+    # We can support forcing a choice to be ordered by tagging it:
+    # (I prefer order over transition(al). at least to me it's clearer.
```

Comment:
what is this comment doing in the code?

Reply:
ah I copy pasted it for myself for reference. will clean it up